x-pack/solutions/observability/plugins/observability_agent_builder/server/routes/ai_insights/get_alert_ai_insights.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest, Logger } from '@kbn/core/server';
+import type { InferenceClient } from '@kbn/inference-common';
+import { MessageRole } from '@kbn/inference-common';
+import dedent from 'dedent';
+import moment from 'moment';
+import type { ObservabilityAgentBuilderDataRegistry } from '../../data_registry/data_registry';
+
+/**
+ * These types are derived from the generated alerts-as-data schemas:
+ * - `AlertSchema` in `kbn-alerts-as-data-utils` (see `alert_schema.ts`) which defines
+ *   `kibana.alert.start` and other technical fields.
+ * - `ObservabilityApmAlertSchema` in `observability_apm_schema.ts`, which adds
+ *   the APM-specific fields like `service.*` and `transaction.*`.
+ *
+ * We only rely on these well-known keys; all other properties are treated as
+ * opaque via the index signature below so this type can safely represent any
+ * Observability alert document.
+ */
+export interface AlertDocForInsight {
+  'service.name'?: string;
+  'service.environment'?: string;
+  'transaction.type'?: string;
+  'transaction.name'?: string;
+  'kibana.alert.start'?: string | number;
+  [key: string]: unknown;
+}
+
+interface GetAlertAiInsightParams {
+  alertDoc: AlertDocForInsight;
+  inferenceClient: InferenceClient;
+  connectorId: string;
+  dataRegistry: ObservabilityAgentBuilderDataRegistry;
+  request: KibanaRequest;
+  logger: Logger;
+}
+
+interface AlertAiInsightResult {
+  summary: string;
+  context: string;
+}
+
+export async function getAlertAiInsight({
+  alertDoc,
+  inferenceClient,
+  connectorId,
+  dataRegistry,
+  request,
+  logger,
+}: GetAlertAiInsightParams): Promise<AlertAiInsightResult> {
+  const relatedContext = await fetchAlertContext({ alertDoc, dataRegistry, request, logger });
+  const summary = await generateAlertSummary({
+    inferenceClient,
+    connectorId,
+    alertDoc,
+    context: relatedContext,
+  });
+
+  return { summary, context: relatedContext };
+}
+
+// Time window offsets in minutes before alert start
+const TIME_WINDOWS = {
+  serviceSummary: 5,
```

Comment:
nit: This looks a bit opaque. It's called "time window" but it doesn't specify a range. Also, just looking at the number `5` it's difficult to determine the unit (5 seconds, 5 minutes, 5 hours?)

Although more repetitive we could consider:

```suggestion
  serviceSummary: {
    start: alertTime.clone().subtract(5, 'minutes').toISOString(),
    end: alertTime.toISOString(),
  },
```

Nothing blocking, just a thought